src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java

```diff
@@ -57,4 +67,59 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
         }
         return sb.toString();
     }
+    /**
+     * Will check the config file to see if players are allowed to use the command that is sent here
+     * and will verify that they are of correct rank to do so
+      *@param player the players/senders name
+     * @param theCommand which command to check if the player can use it
+     * @param colonyId the id of the colony.
+     * @return boolean
+     */
+
+    public boolean canPlayerUseCommand(final EntityPlayer player, final Commands theCommand, final int colonyId)
+    {
+        final Colony chkColony = ColonyManager.getColony(colonyId);
+        if(chkColony == null)
+        {
+            return false;
+        }
+        return canCommandSenderUseCommand(theCommand)
+                && (chkColony.getPermissions().getRank(player).equals(Permissions.Rank.OFFICER) || chkColony.getPermissions().getRank(player).equals(Permissions.Rank.OWNER));
+    }
+    /**
+     * Will check the config file to see if players are allowed to use the command that is sent here
+     *
+     * @param theCommand which command to check if the player can use it
+     * @return boolean
+     */
+    public boolean canCommandSenderUseCommand(Commands theCommand)
+    {
+        switch (theCommand)
+        {
+            case CITIZENINFO: return Configurations.canPlayerUseCitizenInfoCommand;
+            case COLONYTP: return Configurations.canPlayerUseCTPCommand;
+            case KILLCITIZENS: return Configurations.canPlayerUseKillCitizensCommand;
+            case LISTCITIZENS: return Configurations.canPlayerUseListCitizensCommand;
+            case RESPAWNCITIZENS: return Configurations.canPlayerRespawnCitizensCommand;
+            case SHOWCOLONYINFO: return Configurations.canPlayerUseShowColonyInfoCommand;
+            case ADDOFFICER: return Configurations.canPlayerUseAddOfficerCommand;
+            case DELETECOLONY: return Configurations.canPlayerUseDeleteColonyCommand;
+        }
+        return false;
+    }
+
+    /**
+     * Will check to see if play is Opped for the given command name
+     *
+     * @param sender to check the player using the command
+     * @param cmdName the name of the command to be checked
+     * @return boolean
+     */
+    @NotNull
+    public boolean isPlayerOpped(@NotNull final ICommandSender sender, String cmdName)
+    {
+        boolean checkedDone;
+        checkedDone= FMLCommonHandler.instance().getMinecraftServerInstance().canCommandSenderUseCommand(3,cmdName);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)